### ISSUE TITLE
Per-platform python_interpreter settings?

### ISSUE THREAD

**Issue by fredrikaverpil** [DamnWidget/anaconda#605]
Is it possible to somehow define one `python_interpreter` setting on a per-platform basis?

I was thinking something along these lines:

```json
{
    "windows:
    {
        "python_interpreter": "C:/python35/python.exe"
    },
    "osx:
    {
        "python_interpreter": "/usr/bin/python3.5"
    },
    "linux:
    {
        "python_interpreter": "/usr/bin/python3.5"
    },
}
```

And if not, could it be implemented?

I am looking to be able to do this in the Sublime project settings.

**Comment by DamnWidget** [comment]
That is not currently implemented but is interesting

**Comment by MattDMo** [comment]
This would probably be a better format for a `.sublime-project` file:

```js
{
    // ...
    "settings":
    {
        "python_interpreter":
        {
            "windows": "c:/python35/python.exe",
            "osx": "/Library/Frameworks/Python.framework/Versions/3.5/bin/python3.5",
            "linux": "/usr/bin/python3.5"
        },
        // ...
    }
}
```

Each of the keys of the `python_interpretor` dict is the platform-specific output from `sublime.platform()`, so it would be quite easy to know which value to access.

@DamnWidget is the only relevant code in `commands/set_python_interpreter.py`, or are there other files we'd have to change? I'd recommend checking for a simple string value first, so we don't break existing behavior.

**Comment by DamnWidget** [comment]
Function `get_interpreter` in `anaconda_lib/helpers.py` that is being used in `anaconda_lib/workers/interpreter.py` and 'anaconda_lib/workers/market.py`. Then there is a direct access to the settings in `decorators.py` but the function doing it was deprecated long in 2.0.0 and is not being used anymore.

**Comment by j9ac9k** [comment]
Hi @DamnWidget I'm just wanting to +1 this feature request as I use Sublime Text on multiple platforms.  Also, is there a way to check which python interpreter Anaconda is referencing?  I use conda environments, and Anaconda does not seem to use the first `python` that is in the `PATH`.

**Comment by DamnWidget** [comment]
Hi @j9ac9k 
You can easily check with python interpreter is in use writing the following in the ST3 console (I should probably add a command to do it or something)
```python
from Anaconda.anaconda_lib.workers.market import Market
print(Market.lookup(window.id()).interpreter)
```

**Comment by j9ac9k** [comment]
Thanks for the super fast reply.  I suppose I should elaborate a little more, is there a way I can print the full path to the python interpreter?  For example, I currently have `python` specified as the interpreter (and that's all that is returned).  With macOS and homebrew `which -a python` in my terminal returns 4 items.  I suspect Anaconda is using one of those `python`'s but just not the first one.

**Comment by j9ac9k** [comment]
Just a follow up to my issue, which I think is a successful workaround.

I just installed the following package:

https://github.com/Gr1N/SublimeFixMacEnv

And added the correct python path to the environmental variable "PATH" (on my macOS installation), and now Anaconda is behaving properly (it seeing the correct `site-packages` folder and is able to do my import validation and such).

**Comment by DamnWidget** [comment]
Your problem is that Sublime Text 3 in macOS and GNU/Linux does not import the environment variables that yous `$SHELL` defines, so even if in the terminal the `python` in your path is pointing to let's say `/user/local/bin/python` that does not meas that it will point to the same path in your ST3.

**Comment by j9ac9k** [comment]
Right, I got that part, but I was trying to figure out which `python` ST3 is using.

**Comment by DamnWidget** [comment]
Only ST3 can give you an answer to that question, I guess you can do something like this in the ST3 console:
```python
subprocess.check_output(['python', '-c', "import sys; print(sys.executable);"])
```

**Comment by j9ac9k** [comment]
thanks, looking back on the issue, it seems like I hijacked it, my apologies.
